File: ws/models/user_model.py
'''User models'''

from pydantic import BaseModel

class User(BaseModel):
  username:         str
  email:            str
  admin:            int
  created:          str
  activity:         str
  characters:       str

class UserInDB(User):
  hashed_password:  str

class UserRegister(BaseModel):
  username:         str
  password:         str
  email:            str

File: src/app/shared/models/user.model.ts
/**
 * Model for auth response
 */
export interface ITokenUser {
  info: IUser;
}

/**
 * User information
 */
export interface IUser {
  username: string;
  email: string;
  admin: number;
  created: string;
  activity: string;
  characters: string;
}

/**
 * User Registration information
 */
export interface IRegisterUser {
  username: string;
  email: string;
  password: string;
}
